IdentityCore/src/oauth2/aad_base/MSIDAADWebviewFactory.m

```diff
@@ -67,6 +67,17 @@ @implementation MSIDAADWebviewFactory
 
 - (MSIDWebviewSession *)embeddedWebviewSessionFromConfiguration:(MSIDWebviewConfiguration *)configuration customWebview:(WKWebView *)webview context:(id<MSIDRequestContext>)context
 {
+    if (![NSThread isMainThread])
```

Comment:
nit: I think we have this code in multiple places: check if main thread -> perform on main thread if necessary. Could we perhaps benefit from a helper that takes a block perhaps?

Reply:
Not sure if block can bring much value here. We would need to move all code into this block. Let's discuss.